hawtio-util/src/test/java/io/hawt/util/FilesTest.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @PrepareForTest({Files.class, java.nio.file.Files.class})
+  @Test
+  public void getMimeTypeException() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    mockStatic(java.nio.file.Files.class);
+
+    // Arrange
+    when(file.toPath()).thenReturn(Paths.get("/tmp/file"));
+    when(file.isDirectory()).thenReturn(false);
+    when(file.getName()).thenReturn("file");
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenThrow(new IOException());
+
+    // Act & Assert result
+    Assert.assertEquals("text/plain", Files.getMimeType(file));
+  }
+
+  @PrepareForTest({Files.class, java.nio.file.Files.class})
+  @Test
+  public void getMimeType() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    mockStatic(java.nio.file.Files.class);
+
+    // Arrange, Act & Assert result
+    when(file.toPath()).thenReturn(Paths.get("/tmp/file"));
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenReturn("application/octet-stream");
+    Assert.assertEquals("application/octet-stream", Files.getMimeType(file));
+
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenReturn("   ");
+    when(file.isDirectory()).thenReturn(true);
+    Assert.assertEquals("application/zip", Files.getMimeType(file));
+
+    when(file.isDirectory()).thenReturn(false);
+    
+    when(file.getName()).thenReturn(".xml");
+    Assert.assertEquals("application/xml", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".wadl");
+    Assert.assertEquals("application/wadl+xml", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".wsdl");
+    Assert.assertEquals("application/wsdl+xml", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".xsd");
+    Assert.assertEquals("application/xsd+xml", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".json");
+    Assert.assertEquals("application/json", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".html");
+    Assert.assertEquals("application/html", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".htm");
+    Assert.assertEquals("application/html", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".properties");
+    Assert.assertEquals("text/x-java-properties", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".jpg");
+    Assert.assertEquals("image/jpeg", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".jpeg");
+    Assert.assertEquals("image/jpeg", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".png");
+    Assert.assertEquals("image/png", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".gif");
+    Assert.assertEquals("image/gif", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".svg");
+    Assert.assertEquals("image/svg+xml", Files.getMimeType(file));
+
+    when(file.getName()).thenReturn(".txt");
```

Comment:
Add a test like this one but with no extension, to show that `"text/plain"` is the default case